### ISSUE TITLE
ArgumentError: invalid value for BigDecimal(): ""

### ISSUE THREAD

**Issue by moiristo** [Barzahlen/camt_parser#21]
Hi,

Somehow a client of us managed to get hold of a CAMT file with transactions without an amount. This yields the error described with the trace below. As this is not easy to catch, I think the gem should check the value before building the BigDecimal. Maybe a validation? I'm using `amount_in_cents` now, as this yields zero when no amount is set.

```
[GEM_ROOT]/gems/camt_parser-2.4.0/lib/camt_parser/misc.rb:9 :in `BigDecimal`
[GEM_ROOT]/gems/camt_parser-2.4.0/lib/camt_parser/misc.rb:9 :in `to_amount`
[GEM_ROOT]/gems/camt_parser-2.4.0/lib/camt_parser/general/transaction.rb:11 :in `amount`
```

**Comment by tobischo** [comment]
Hmm, yeah, unfortunately way too many fields in camt files are actually optional.

What would your expectation be in terms of behaviour?
I could imagine both: a better exception or defaulting to 0, which is technically correct for something that has no amount.

Could you provide us with a little bit of background on the reason for that transaction not having an amount? What kind of transaction is it?

**Comment by moiristo** [comment]
I wish I knew, but unfortunately I can't see what the user has uploaded. However, I'd say that the most logical behaviour would be that `amount_in_cents` and `amount` both yield `nil` when no value is set, as no amount has been passed?

**Comment by tobischo** [comment]
Yeah, that's true.
We will adapt it accordingly

**Comment by tobischo** [comment]
Resolved in #22 and released as v2.6.1

**Comment by moiristo** [comment]
Thanks!
